library/Icingadb/Web/Controller.php

```diff
@@ -550,6 +550,86 @@ public function export(Query ...$queries)
             return true;
         }
 
+        if ($this->format === 'json' || $this->format === 'csv') {
+            $isJsonFormat = $this->format === 'json';
+            $response = $this->getResponse();
+
+            $fileName = $this->getRequest()->getControllerName();
+            if ($isJsonFormat) {
+                $response
+                    ->setHeader('Content-Type', 'application/json')
+                    ->setHeader('Cache-Control', 'no-store')
+                    ->setHeader(
+                        'Content-Disposition',
+                        'inline; filename=' . $fileName . '.json'
+                    )
+                    ->sendResponse();
+            } else {
+                $response
+                    ->setHeader('Content-Type', 'text/csv')
+                    ->setHeader('Cache-Control', 'no-store')
+                    ->setHeader(
+                        'Content-Disposition',
+                        'attachment; filename=' . $fileName . '.csv'
+                    )
+                    ->sendResponse();
+            }
+            // Disable output buffering to stream large amounts of data without a memory exhausted error.
```

Comment:
Not sure about the usefulness of this comment to be honest, but anyway, add an empty line beforehand. There's a control structure above it after all.